constance/test/unittest.py

```diff
@@ -44,9 +45,19 @@ def modify_test_case(self, test_case):
         original_pre_setup = test_case._pre_setup
         original_post_teardown = test_case._post_teardown
 
-        def _pre_setup(inner_self):
-            self.enable()
-            original_pre_setup(inner_self)
+        if DJANGO_VERSION < (5, 2):
+
+            def _pre_setup(inner_self):
+                self.enable()
+                original_pre_setup(inner_self)
+        else:
+
+            @classmethod
```

Comment:
Can we create a decorator for that so that we don't have to duplicate the code of the method?

Reply:
The decorator is not the only difference, the call to original_pre_setup also differs (as we don't have to pass the instance anymore)
Additionally, this version is probably easier to clean up when Django5.2 will be the minimum supported version